README.md

```diff
@@ -79,6 +80,25 @@ This will authorize all requests with credentials where the username begins with
 `'A'` and the password begins with `'secret'`. In an actual application you would
 likely look up some data instead ;-)
 
+### Timing attacks
+
+When using a custom validation callback, even when comparing hashes, make sure
+to **avoid standard string compare (`==`)**. Use a "constant time compare
+function". This protects you against [timing
+attacks](https://en.wikipedia.org/wiki/Timing_attack).
+
+For example, using
+[safe-compare](https://www.npmjs.com/package/safe-compare):
+
+```js
+const compare = require('safe-compare');
+app.use(basicAuth( { authorizer: myAuthorizer } ))
+
+function myAuthorizer(username, password) {
+    return compare("admin", username) & compare("admin-password", password);
```

Comment:
Looks like there's a typo in this code. I believe @hraban meant a `&&`

```suggestion
    return compare("admin", username) && compare("admin-password", password);
```

Reply:
I initially made the same mistake, but someone pointed out to me that using a short-circuiting conjunction operator (`&&`) will information about the username. Using an eager operator (`&`) will be actually constant time.

Reply:
🤦‍♂️
yes let's document to prevent same mistakes in the future! ;-)